tests/test_tabbed_content.py

```diff
@@ -696,3 +696,63 @@ def compose(self) -> ComposeResult:
         tabbed_content.show_tab(tab_id)
         await pilot.pause()
         assert tabbed_content.active == tab_id
+
+
+async def test_disabling_nested_tabs():
+    """Regression test for https://github.com/Textualize/textual/issues/3145."""
+
+    class TabbedApp(App):
+        def compose(self) -> ComposeResult:
+            with TabbedContent(id="tabbed-content"):
+                with TabPane("Tab Pane 1", id="first-tab"):
+                    yield Label("foo")
+                with TabPane("Tab Pane 2", id="second-tab"):
+                    yield Label("bar")
+                with TabPane("Tab Pane 3", id="third-tab"):
+                    with TabbedContent(id="another-tabbed-content"):
+                        with TabPane("Inner Pane 1", id="inner-tab-1"):
```

Comment:
Would it not make more sense to test where the ID is duplicated, so `first-tab` is used again for the nested first tab?

Reply:
I took the regression test from the issue and rolled with it because it looked good to me.
What you're suggesting is that change the IDs of the inner tab panes to match the outer panes?

Reply:
Aye, that's it. A worst case is going to be the same ID being used within a `TabPane` that is within a `TabPane` of the same ID. This should then hit all the bases.

Reply:
Thanks for the suggestion. Implemented in 22b63f6.